v3/lints/cabf_ev/lint_cabf_org_identifier_psd_vat_has_state.go

```diff
@@ -0,0 +1,57 @@
+/*
+ * ZLint Copyright 2024 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package cabf_ev
+
+import (
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/v3/lint"
+	"github.com/zmap/zlint/v3/util"
+)
+
+func init() {
+	lint.RegisterCertificateLint(&lint.CertificateLint{
+		LintMetadata: lint.LintMetadata{
+			Name:          "e_cabf_org_identifier_psd_vat_has_state",
+			Description:   "The cabfOrganizationIdentifier field for PSD org VAT Registration Schemes cannot include the referenceStateOrProvince field.",
+			Citation:      "Current PSD and VAT based identifiers are issued from Country level only. No state or province information is included.",
```

Comment:
The citation is usually the section/paragraph/clause number of the given requirement (E.G `7.1.1.2a`).

Would you mind finding that and attaching it here? Its purpose is because the `Description` field is usually a paraphrasing and it is important for others to be able to quickly lookup the requirement in case a dispute arises.

Indeed, I tried to grep around in [CABF EV 1.7.0](https://cabforum.org/uploads/CA-Browser-Forum-EV-Guidelines-v1.7.0.pdf) in order to independently verify this lint but I couldn't find it's source (I assume that it is there, but it's important that these are fast and easy to look up).

Reply:
I've instead added the section, which is 7.1.4.2.8. 

The problem with this one is that it's not very clearly outlined. Section 7.1.2.2 is what this lint is based on, yet the definition of what goes where, is indirectly in 7.1.4.2.8, which shows that only NTR based orgIdentifiers have the state value. 

Let me know if you see a better way of showing this within the code

Reply:
I apologize deeply, but I'm a bit confused about the citation numbers.

I certainly see this requirement in the [Ballot SC17](https://cabforum.org/2019/05/21/ballot-sc17-version-7-alternative-registration-numbers-for-ev-certificates/#motion-begins).

However, the entirety of section 7 in the EV guidelines is just two pages of document preamble.
![image](https://github.com/zmap/zlint/assets/8518514/e3a0371f-1b07-47da-9798-661c4edb74c1)

I am quite sure that I'm staring something blank in the face, but I'm not seeing it.

Reply:
@christopher-henderson I just now see you're on V1.7.0. The EVGs were recently converted to the RFC3647 format, hence, I've used the current version's section numbers as reference:

https://github.com/cabforum/servercert/blob/main/docs/EVG.md#71428-subject-organization-identifier-field

For version 1.7.0, the relevant section number is 9.2.8. 

(This has me wondering, which numbering should be used, the section-at-time-of-ballot, or section-at-time-of-lint-creation?)

Reply:
Thank you for pointing me in the right direction @XolphinMartijn!

> The EVGs were recently converted to the RFC3647 format

This may end up being problematic should CABF halt publishing they way that they have been all these years. However, ZLint generally makes references to the older format, so if you would not mind then I would appreciate that we cite `9.2.8`.

---

Since this lint is paraphrasing, I presume that this is the requirement that you are citing:

> For the NTR Registration Scheme identifier, if required under [Section 7.1.4.2.4](https://github.com/cabforum/servercert/blob/main/docs/EVG.md#71424-subject-jurisdiction-of-incorporation-or-registration-field), a 2 character ISO 3166-2 identifier for the subdivision (state or province) of the nation in which the Registration Scheme is operated, preceded by plus "+" (0x2B (ASCII), U+002B (UTF-8));

That is, NTR is the only scheme that explicitly allows the `State` filed. Thus, this lint derives the inverse that a `State` is not legal for either `VAT` nor `PSD` schemes, correct?

Reply:
>This may end up being problematic should CABF halt publishing they way that they have been all these years. 

Should the question arrise, I will advocate the need to keep old versions around ;)

>However, ZLint generally makes references to the older format, so if you would not mind then I would appreciate that we cite 9.2.8.

Sure. Done!

>That is, NTR is the only scheme that explicitly allows the State filed. Thus, this lint derives the inverse that a State is not legal for either VAT nor PSD schemes, correct?

That is correct indeed.